### PR TITLE
feat(membership): add membership adapter

### PR DESCRIPTION
**Pull request by ZixiXu**
Added the interface of MembershipAdapter

### CODE REVIEW

File: src/MembershipAdapter.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import {throwError} from 'rxjs';
+
+import WebexAdapter from './WebexAdapter';
+
```
Comment (lalli-flores) [inline src/MembershipAdapter.js:4]: Can you add the type definition of a membership here?
We do something similar with [people](https://github.com/webex/component-adapter-interfaces/blob/master/src/PeopleAdapter.js#L5) and [meeting](https://github.com/webex/component-adapter-interfaces/blob/master/src/MeetingsAdapter.js#L5)

File: src/MembershipAdapter.test.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import {isObservable} from 'rxjs';
+
+import MembershipAdapter from './MembershipAdapter';
+
+describe('Membership Adapter Interface', () => {
```
Comment (lalli-flores) [inline src/MembershipAdapter.test.js:5]:
```suggestion
describe('Memberships Adapter Interface', () => {
```

File: src/MembershipAdapter.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * Enum for meeting types of membership.
+ *
+ * @readonly
+ * @enum {string}
+ */
+export const MembershipType = {
```
Comment (lalli-flores) [inline src/MembershipAdapter.js:11]:
```suggestion
export const DestinationType = {
```

File: src/MembershipsAdapter.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+   * will emit a new updated Membership object, if datasource permits.
+   *
+   * @param {string} destinationID  ID of the destination for which to get members.
+   * @param {DestinationType} destinationType  type of the membership destination.
+   * @returns {Observable.<Membership>}
+   * @memberof MembershipAdapter
+   */
+  getMembersFromDestination(destinationID, destinationType) {
+    return throwError(
+      new Error('getMembersFromDestination(destinationID, destinationType) must be defined in MembershipsAdapter')
```
[the remark this answers is not in the document]
Comment (ZixiXu) [reply src/MembershipsAdapter.js:46]: This is a variable as the argument of throwError(), I think we cannot add a semi-colon here. You can see this example:
https://github.com/webex/component-adapter-interfaces/blob/master/src/PeopleAdapter.js#L69

The semi-colon is already added in line 47.

[the remark this answers is not in the document]
Comment (lalli-flores) [reply src/MembershipsAdapter.js:46]: haha yes, I see it 🤦🏻‍♀️
You can ignore me 😛

**Pull request merged**